### ISSUE TITLE
self.get(..) has to be the first thing to be yielded in a Process

### ISSUE THREAD

**Issue by mkmik** [erikkaplun/spinoff#1]
``` python
from spinoff.actor import Actor
from spinoff.actor.process import Process
from spinoff.util.logging import dbg
from spinoff.util.async import sleep


class ExampleProcess(Process):
    def run(self):
        #_ = yield self.get() # if we first get() something, then it would work
        yield sleep(1.0)

        while True:
            something = yield self.get()
            dbg("got something %r" % (something,))

class Test(Actor):
    def pre_start(self):
        child = self.spawn(ExampleProcess)
        child << "blah"
        child << "again"

    def receive(self, msg):
        pass
```

It outputs:

```
2012-11-20 23:33:29+0100 [-] Log opened.
2012-11-20 23:33:29+0100 [-] twistd 12.2.0 (/home/marko/.virtualenvs/spinoff/bin/python 2.7.3) starting up.
2012-11-20 23:33:29+0100 [-] reactor class: twisted.internet.epollreactor.EPollReactor.
22:33:29 18338 log    events.py:129 Events    in log: Running: example.Test 
22:33:29 18338 log    events.py:129 Events    in log: No remoting requested; specify `--remoting/-r <nodeid>` (nodeid=host:port) to set up remoting 
22:33:29 18338 log    events.py:129 Events    in log: Spawned </test$1> 
22:33:30 18338 fail   _actor.py:1215 cell:ExampleProcess@/test$1/exampleprocess$1  N ?  in failure report 

 Traceback (most recent call last):
  File "/home/marko/Projects/spinoff/spinoff/actor/_actor.py", line 918, in _process_messages
    yield self._process_one_message(message)
  File "/home/marko/Projects/spinoff/spinoff/actor/_actor.py", line 939, in _process_one_message
    yield self._do_start()
  File "/home/marko/Projects/spinoff/spinoff/actor/_actor.py", line 1020, in _do_start
    yield self._construct()
  File "/home/marko/Projects/spinoff/spinoff/actor/_actor.py", line 1010, in _construct
    raise CreateFailed("Actor failed to start", actor)
CreateFailed: ('Actor failed to start', <proc-impl:ExampleProcess@/test$1/exampleprocess$1>)

-------
CAUSE:

      Traceback (most recent call last):
        File "/home/marko/Projects/spinoff/spinoff/actor/_actor.py", line 1007, in _construct
    del self._ongoing
      AttributeError: _ongoing

22:33:30 18338 fail   events.py:129 Events    in log: Error(</test$1/exampleprocess$1>, CreateFailed('Actor failed to start', <proc-impl:ExampleProcess@/test$1/exampleprocess$1>, AttributeError('_ongoing',)), <traceback object at 0x25f3830>) 
22:33:30 18338 debug  example.py:16 proc-impl:ExampleProcess@/test$1/exampleprocess$1    in run: got something 'blah' 
22:33:30 18338 debug  example.py:16 proc-impl:ExampleProcess@/test$1/exampleprocess$1    in run: got something 'again' 
22:33:30 18338 debug  _actor.py:1030 cell:Test@/test$1  X  in sup: CreateFailed('Actor failed to start', <proc-impl:ExampleProcess@/test$1/exampleprocess$1>, AttributeError('_ongoing',)) ← </test$1/exampleprocess$1>      
22:33:30 18338 debug  _actor.py:1043 cell:Test@/test$1  X  in sup: → Stop → </test$1/exampleprocess$1>  
```

**Comment by mkmik** [comment]
I'm wondering why do you need to keep track of ongoing `pre_start` using the same `self._ongoing` attr used for keeping track of ongoing gets.

I tried this and it appears to fix this problem:

``` diff
--- a/spinoff/actor/_actor.py
+++ b/spinoff/actor/_actor.py
@@ -1002,9 +1002,7 @@ class Cell(_BaseCell):
             pre_start = actor.pre_start
             args, kwargs = actor._Actor__startargs
             try:
-                self._ongoing = pre_start(*args, **kwargs)
-                yield self._ongoing
-                del self._ongoing
+                yield pre_start(*args, **kwargs)
             except Exception:
                 # dbg(u"☹")
                 raise CreateFailed("Actor failed to start", actor)
```

although I have no idea on the impact on the rest of the framework, since unit tests are failing anyways (both in HEAD and in v0.5.2). Is this normal or there is something weird in my setup?

**Comment by erikkaplun** [comment]
I have to fix the unit tests to start with...

But anyway—your proposed change will break at least one thing that I can think of. Afaik, `Actor`/`Cell` is quite OK but how `Process` has been made work on top of `Actor`/`Cell` is kind of ugly right now and that's probably where the problem lies. This is not to mention the hunger for a refactoring in the core message processing logic of `Cell`.

**Comment by erikkaplun** [comment]
Seems to have been fixed in 8e96616; feel free to re-open if not.
